lib/subjects/seen_remover.rb

```diff
@@ -0,0 +1,36 @@
+module Subjects
+  class SeenRemover
+
+    attr_reader :user, :workflow, :append_sms_ids
+
+    def initialize(user, workflow, append_ids)
+      @user = user
+      @workflow = workflow
+      @append_sms_ids = append_ids
```

Comment:
I refactored to inject the user seens dependency but I felt it was cleaner to use the append ids in the initializer. If the need to reuse the object with new append ids arises we should refactor this out.